content/blog/2020/07/2020-07-27-repository-signing-keys-changing.adoc

```diff
@@ -0,0 +1,82 @@
+The Linux repositories that deliver the weekly release were updated with new GPG keys with the release of Jenkins 2.232.
+
+Beginning with Jenkins LTS release link:/changelog-stable/#v2.235.3[2.235.3], stable repositories will be signed with the same GPG keys that sign the weekly repositories.
+Administrators of Linux systems *must* install the new signing keys on their Linux servers *before* installing Jenkins 2.235.3.
+
+== Debian/Ubuntu
+
+Update Debian compatible operating systems (Debian, Ubuntu, Linux Mint Debian Edition, etc.) with the command:
+
+.Debian
+----
+# wget -qO - https://pkg.jenkins.io/debian-stable/jenkins.io.key | apt-key add -
+----
+
+== Red Hat/CentOS
+
+Update Red Hat compatible operating systems (Red Hat Enterprise Linux, CentOS, Fedora, Oracle Linux, Scientific Linux, etc.) with the command:
+
+.Red Hat
+----
+# rpm --import https://pkg.jenkins.io/redhat-stable/jenkins.io.key
+----
+
+////
+Need to confirm that openSUSE and SUSE do not require an update to the repository signing keys.
+If an update is needed, then need that command here.
+////
+== Frequently Asked Questions
+
+=== What if I don't update the repository signing key?
+
+Updates will be *blocked* by the operating system package manager (apt, yum, dnf) on operating systems that have not installed the new repository signing key.
+Sample messages from the operating system may look like:
+
+.Debian
+----
+Reading package lists... Done
+W: GPG error: https://pkg.jenkins.io/debian-stable binary/ Release:
+    The following signatures couldn't be verified because the public key is not available:
+        NO_PUBKEY FCEF32E745F2C3D5
+E: The repository 'https://pkg.jenkins.io/debian-stable binary/ Release' is not signed.
+N: Updating from such a repository can't be done securely, and is therefore disabled by default.
+N: See apt-secure(8) manpage for repository creation and user configuration details.
+----
+
+.Red Hat
+----
+Downloading packages:
+warning: /var/cache/yum/x86_64/7/jenkins/packages/jenkins-2.235.3-1.1.noarch.rpm:
+    Header V4 RSA/SHA512 Signature, key ID 45f2c3d5: NOKEY
+Public key for jenkins-2.235.3-1.1.noarch.rpm is not installed
+----
+
+=== Why is the repository signing key being updated?
+
+The original repository GPG signing key is owned by Kohsuke Kawaguchi.
+Rather than require that Kohsuke disclose his personal GPG signing key, the core release automation project has used a new repository signing key.
+The updated GPG repository signing key is used in the weekly repositories and the stable repositories.
+
+=== Which operating systems are affected?
+
+Operating systems that use Debian package management (apt) and operating systems that use Red Hat package management (yum and dnf) need the new repository signing key.
+
+Other operating systems like Windows, macOS, FreeBSD, OpenBSD, Solaris, and OpenIndiana are not affected.
```

Comment:
There’s the new windows msi as well does that need mentioning?

Reply:
> There’s the new windows msi as well does that need mentioning?

Windows will accept an MSI so long as it is signed and (I believe) the signature is valid and not expired.

I frame that mentally as a "code signing" case rather than a "repository signing" case.  The code signing on jenkins.war and on jenkins.msi have both been updated but the signing is rarely checked by users.  That mental model is flawed because they are both forms of code signing, but the separation helps me describe them differently.

The jenkins.war and the jenkins.msi are signed with a code signing certificate, while the deb file and the rpm file are signed with a GPG private key.  The GPG public key needs to be installed into the keyring of the receiving system in order to validate the signing.  The code signing certificate validation does not require any additional installation by the user.

We had no reports of any concerns with the new code signing of jenkins.war in the weekly transition to core release automation.  We've had no reports about the certificate that expired July 22, 2020 that is used to sign the LTS jenkins.war.

We've had several reports about the GPG signing used for the deb repository and the rpm repository and their contents.

I'm sure I've made mistakes in my description above, though I believe the mistakes I've made are unlikely to change the actions that users need to take.

We could describe the change of code signing in this blog post, though I worried that it might confuse readers, since they must take action on the GPG signing key change and no action is needed on the war signing key change.

Reply:
Sure only small correction is that code signing is a bit different to TLS certs, they are valid as long as when it was signed the cert was valid, I think there’s an exception that if the root expires then it needs resigning.

Reply:
Thanks.  If time allows, I'll try to add another "Frequently Asked Question" that uses a "question" as the excuse to state that the MSI and the WAR file are both using new code signing certificates as well.  That may help the google search results ranking and may help a few readers if they have questions.

Reply:
Yes, I believe code signing for these packages needs to be referenced as well

Reply:
Good point.  Seems like the changelog should also mention that the MSI installer is changing to the new version.  Jenkins 2.235.2 delivered the Windows installer that is 100+ MB, while Jenkins 2.235.3 will deliver an installer that is less than 75 MB.

I'll update to include that.

Also need to mention that this the LTS release that officially stops delivering an osx installer.

Reply:
MSI reference has been added in 69b08beeb9f9cf1e859c2f3580eec5920a3333c8